src/Hamiltonians/Molecular.jl

```diff
@@ -0,0 +1,790 @@
+"""
+    MolecularHamiltonian(
+        fcidump_path::String;
+        starting_address::Union{Nothing,FermiFS2C}=nothing,
+        specifier::String="",
+    ) <: AbstractHamiltonian
+    MolecularHamiltonian(
+        fd::ElemCo.FciDumps.FDump;
+        starting_address::Union{Nothing,FermiFS2C}=nothing,
+        specifier::String="",
+    ) <: AbstractHamiltonian
+
+Implements an electronic ab-initio Hamiltonian based on electron overlap integrals in
+FCIDUMP format. It can be used to describe the electronic structure of a molecule
+with fixed nuclear positions (i.e. under the Born-Oppenheimer approximation).
+
+The expression for the Hamiltonian is
+```math
+\\begin{aligned}
+    Ĥ  & = h₀ + Ĥ₁ + Ĥ₂, \\\\
+    Ĥ₁ & = \\sum_{i,j,σ} h_{ij}  a^†_{j,σ} a_{i,σ}, \\\\
+    Ĥ₂ & = ½ \\sum_{i,j,σ,τ} V_{ijkl} a^†_{i,σ} a^†_{j,τ} a_{l,τ} a_{k,σ}.
```

Comment:
I still found mistakes in the formula.
```suggestion
    Ĥ₁ & = \\sum_{i,j,σ} h_{ij}  a^†_{i,σ} a_{j,σ}, \\\\
    Ĥ₂ & = ½ \\sum_{i,j,k,l,σ,τ} V_{ijkl} a^†_{i,σ} a^†_{j,τ} a_{l,τ} a_{k,σ}.
```
Please double check that they are correct. It is very confusing to have incorrect formulas in the documentation.

Reply:
Maybe we should align the notation of subscript with indices of `fcidump.int2` and mode indices of `excitation` used in the code. I have updated it in the latest commit for a test.

Reply:
OK.